tracer/samples/RabbitMQ.DistributedTracing/Receive/Receive.cs

```diff
@@ -31,61 +35,27 @@ public static void Main()
                         var body = ea.Body.ToArray();
                         var message = Encoding.UTF8.GetString(body);
 
-                        // Read the basic property headers and extract the Datadog properties
-                        var headers = ea.BasicProperties.Headers;
-                        ulong? parentSpanId = null;
-                        ulong? traceId = null;
-                        SamplingPriority? samplingPriority = null;
-
-                        // Parse parentId header
-                        if (headers?[HttpHeaderNames.ParentId] is byte[] parentSpanIdBytes)
-                        {
-                            parentSpanId = BitConverter.ToUInt64(parentSpanIdBytes, 0);
-                        }
+                        // Log message and properties to screen
+                        Console.WriteLine(" [x] Received.");
+                        Console.WriteLine("     Message: {0}", message);
 
-                        // Parse traceId header
-                        if (headers?[HttpHeaderNames.TraceId] is byte[] traceIdBytes)
+                        // extract the context to this message as it has been lost when enqueuing
+                        var messageHeaders = ea.BasicProperties?.Headers;
+                        if (messageHeaders != null)
                         {
-                            traceId = BitConverter.ToUInt64(traceIdBytes, 0);
-                        }
+                            var spanContextExtractor = new SpanContextExtractor();
+                            var parentContext = spanContextExtractor.Extract(messageHeaders, GetHeaderValues);
+                            var spanCreationSettings = new SpanCreationSettings() {Parent = parentContext};
 
-                        // Parse samplingPriority header
-                        if (headers?[HttpHeaderNames.SamplingPriority] is byte[] samplingPriorityBytes)
-                        {
-                            var samplingPriorityString = Encoding.UTF8.GetString(samplingPriorityBytes);
-                            if (Enum.TryParse<SamplingPriority>(samplingPriorityString, out var result))
+                            // Create child spans
+                            using var scope = Tracer.Instance.StartActive("child.span", spanCreationSettings);
                             {
-                                samplingPriority = result;
+                                Console.WriteLine("You can safely add t");
```

Comment:
I think we should keep the previous messages, just to demonstrate it's working. There's a question of whether they should add the amqp tags or not I guess? 🤔 

```suggestion
                                Console.WriteLine("You can safely add t");
                                Console.WriteLine("     Active TraceId: {0}", scope.Span.TraceId);
                                Console.WriteLine("     Active SpanId: {0}", scope.Span.SpanId);
                                
                                // Do work inside the Datadog trace
                                Thread.Sleep(1000);
```